mlxtend/frequent_patterns/tests/test_fpbase.py

```diff
@@ -0,0 +1,123 @@
+# Sebastian Raschka 2014-2019
+# mlxtend Machine Learning Library Extensions
+# Author: Sebastian Raschka <sebastianraschka.com>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from mlxtend.frequent_patterns import apriori
+from numpy.testing import assert_array_equal
+from mlxtend.utils import assert_raises
+import pandas as pd
+
+
+# need to wrap with this class so FPTestBase is not run as its own unit test
+class FPTestBase(object):
+    def setUp(self):
+        self.one_ary = np.array(
+            [[0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1],
+                [0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1],
+                [1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1],
+                [0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0]])
+
+        self.cols = ['Apple', 'Corn', 'Dill', 'Eggs', 'Ice cream',
+                     'Kidney Beans', 'Milk',
+                     'Nutmeg', 'Onion', 'Unicorn', 'Yogurt']
+
+        self.df = pd.DataFrame(self.one_ary, columns=self.cols)
+
+        self.fpalgo = apriori

```

Comment:
this line is not really needed, right?

Reply:
Yeah, I think you are right.

However, if I remove it, it creates red error lines in my editor because it's not defined (but it will still work because it gets defined).

I guess the best way to handle it may be to force the argument in the constructor:

`test_fpbase.py`
```
    def setUp(self, fpalgo):
        self.one_ary = np.array(
            [[0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1],
                [0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1],
                [1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0],
                [0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1],
                [0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0]])

        self.cols = ['Apple', 'Corn', 'Dill', 'Eggs', 'Ice cream',
                     'Kidney Beans', 'Milk',
                     'Nutmeg', 'Onion', 'Unicorn', 'Yogurt']

        self.df = pd.DataFrame(self.one_ary, columns=self.cols)

        self.fpalgo = fpalgo
```

`test_fpgrowth.py`
```
class FPTestGrowth(unittest.TestCase, FPTestBase):
    def setUp(self):
        FPTestBase.setUp(self, fpgrowth)
```

`test_apriori.py`
```
class FPTestApriori(unittest.TestCase, FPTestBase):
    def setUp(self):
        FPTestBase.setUp(self, apriori)
```